src/main/telemetry/crsf.c

```diff
@@ -59,12 +59,72 @@
 
 #define CRSF_CYCLETIME_US                   100000 // 100ms, 10 Hz
 #define CRSF_DEVICEINFO_VERSION             0x01
-#define CRSF_DEVICEINFO_PARAMETER_COUNT     255
+#define CRSF_DEVICEINFO_PARAMETER_COUNT     0
+
+#define CRSF_MSP_BUFFER_SIZE 96
 
 static bool crsfTelemetryEnabled;
 static bool deviceInfoReplyPending;
 static uint8_t crsfFrame[CRSF_FRAME_SIZE_MAX];
 
+#if defined(USE_MSP_OVER_TELEMETRY)
+typedef struct mspFlatBuffer_s {
+    uint8_t buffer[CRSF_MSP_BUFFER_SIZE];
+    sbuf_t *head; // read-only
+    sbuf_t *tail; // write-only
+} mspFlatBuffer_t;
+
+static mspFlatBuffer_t mspFlatRxBuffer;
+static sbuf_t mspRxBufHead;
+static sbuf_t mspRxBufTail;
+
+void initCrsfMspBuffer(void)
+{
+    uint8_t *headStart = (uint8_t*)&mspFlatRxBuffer.buffer;
+    uint8_t *tailEnd = (uint8_t*)&mspFlatRxBuffer.buffer+CRSF_MSP_BUFFER_SIZE;
+    mspFlatRxBuffer.head = sbufInit(&mspRxBufHead, headStart, headStart);
+    mspFlatRxBuffer.tail = sbufInit(&mspRxBufTail, headStart, tailEnd);
+}
+
+bool bufferCrsfMspFrame(uint8_t *frameStart, int frameLength)
+{
+    bool frameBuffered = false;
+    if(sbufBytesRemaining(mspFlatRxBuffer.tail)>=frameLength) {
+        sbufWriteU8(mspFlatRxBuffer.tail, (uint8_t)frameLength);
+        sbufWriteData(mspFlatRxBuffer.tail, frameStart, frameLength);
```

Comment:
IMO sbufWriteData advances buffer pointers ...

Reply:
Interesting.  A few weeks back it was not.  Must've been corrected.

Reply:
I must've had this confused with ReadData, which does not.

Reply:
`sbufReadData` should move pointer too. It is a bug, I'll fix it ...